Collation errors for postgres clusters created with different locale

When changing my test machine, I mistakenly created a postgres cluster with `SQL_ASCII` encoding (C). This caused SQLancer to throw a lot of errors around collation.

```
$ java -jar sqlancer-*.jar --username ubuntu postgres 2>&1 | grep "^Caused by" | cut --bytes=-95 | sort | uniq -c | sort -nr
     18 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_AT.utf8" for encoding "SQL_A
     15 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_IT.utf8" for encoding "SQL_A
     13 Caused by: org.postgresql.util.PSQLException: ERROR: collation "en_US.utf8" for encoding "SQL_A
     13 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_BE.utf8" for encoding "SQL_A
     12 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_DE.utf8" for encoding "SQL_A
     11 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_LI.utf8" for encoding "SQL_A
     10 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_LU.utf8" for encoding "SQL_A
      8 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_CH.utf8" for encoding "SQL_A
```

However, it was interesting to note that disabling collations only reduce the error, and not completely make it go away.

```
$ java -jar sqlancer-*.jar --username ubuntu postgres --test-collations=false 2>&1 | grep "^Caused by" | cut --bytes=-95 | sort | uniq -c | sort -nr
     67 Caused by: org.postgresql.util.PSQLException: ERROR: invalid byte sequence for encoding "UTF8":
      7 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_IT.utf8" for encoding "SQL_A
      5 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_BE.utf8" for encoding "SQL_A
      4 Caused by: org.postgresql.util.PSQLException: ERROR: collation "en_US.utf8" for encoding "SQL_A
      4 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_LI.utf8" for encoding "SQL_A
      4 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_CH.utf8" for encoding "SQL_A
      3 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_LU.utf8" for encoding "SQL_A
      3 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_DE.utf8" for encoding "SQL_A
      2 Caused by: org.postgresql.util.PSQLException: ERROR: collation "de_AT.utf8" for encoding "SQL_A
```

--- a/src/sqlancer/postgres/PostgresProvider.java
+++ b/src/sqlancer/postgres/PostgresProvider.java
@@ -325,17 +325,24 @@ public class PostgresProvider extends SQLProviderAdapter<PostgresGlobalState, Po
     private String getCreateDatabaseCommand(PostgresGlobalState state) {
         StringBuilder sb = new StringBuilder();
         sb.append("CREATE DATABASE " + databaseName + " ");
-        if (Randomly.getBoolean() && ((PostgresOptions) state.getDbmsSpecificOptions()).testCollations) {
+        if (((PostgresOptions) state.getDbmsSpecificOptions()).testCollations) {
             if (Randomly.getBoolean()) {
-                sb.append("WITH ENCODING '");
-                sb.append(Randomly.fromOptions("utf8"));
-                sb.append("' ");
-            }
-            for (String lc : Arrays.asList("LC_COLLATE", "LC_CTYPE")) {
-                if (!state.getCollates().isEmpty() && Randomly.getBoolean()) {
-                    sb.append(String.format(" %s = '%s'", lc, Randomly.fromList(state.getCollates())));
+                if (Randomly.getBoolean()) {
+                    sb.append("WITH ENCODING '");
+                    sb.append(Randomly.fromOptions("utf8"));
+                    sb.append("' ");
                 }
+                for (String lc : Arrays.asList("LC_COLLATE", "LC_CTYPE")) {
+                    if (!state.getCollates().isEmpty() && Randomly.getBoolean()) {
+                        sb.append(String.format(" %s = '%s'", lc, Randomly.fromList(state.getCollates())));
+                    }
+                }
+                sb.append(" TEMPLATE template0");
             }
+        } else {
+            sb.append("WITH ENCODING '");
+            sb.append(Randomly.fromOptions("utf8"));
+            sb.append("' ");
             sb.append(" TEMPLATE template0");
         }
         return sb.toString();

--- a/src/sqlancer/postgres/PostgresProvider.java
+++ b/src/sqlancer/postgres/PostgresProvider.java
@@ -340,10 +340,7 @@ public class PostgresProvider extends SQLProviderAdapter<PostgresGlobalState, Po
                 sb.append(" TEMPLATE template0");
             }
         } else {
-            sb.append("WITH ENCODING '");
-            sb.append(Randomly.fromOptions("utf8"));
-            sb.append("' ");
-            sb.append(" TEMPLATE template0");
+            sb.append("WITH ENCODING 'UTF8' TEMPLATE template0");
         }
         return sb.toString();
     }

